frontend/src/components/Milestones.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  faCalendar,
+  faFolderOpen,
+  faFire,
+  faCircleCheck,
+  faCircleExclamation,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { useRouter } from 'next/navigation'
+import React from 'react'
+import { ProjectMilestonesType } from 'types/project'
+import { formatDate } from 'utils/dateFormatter'
+import AnchorTitle from './AnchorTitle'
+import ItemCardList from './ItemCardList'
+import { TruncatedText } from './TruncatedText'
+
+interface ProjectMilestonesProps {
+  data: ProjectMilestonesType[]
+  showAvatar?: boolean
+  openMilestones?: boolean
+}
+
+const Milestones: React.FC<ProjectMilestonesProps> = ({
+  data,
+  showAvatar = true,
+  openMilestones,
+}) => {
+  const router = useRouter()
+
+  return (
+    <ItemCardList
+      title={
+        <div className="flex items-center gap-2">
+          <AnchorTitle
+            title={openMilestones ? 'Open Milestones' : 'Closed Milestones'}
+            className="flex items-center leading-none"
+          />
+        </div>
+      }
+      data={data}
+      showAvatar={showAvatar}
+      icon={faFire}
+      renderDetails={(item) => (
+        <div className="mt-2 flex flex-wrap items-center text-sm text-gray-600 dark:text-gray-400">
+          <div className="mr-4 flex items-center">
+            <FontAwesomeIcon icon={faCalendar} className="mr-2 h-4 w-4" />
+            <span>{formatDate(item.createdAt)}</span>
+          </div>
+          <div className="mr-4 flex items-center">
+            <FontAwesomeIcon icon={faCircleCheck} className="mr-2 h-4 w-4 text-green-500" />
```

Comment:
Let's have them in the same grey color as the rest of the icons in this card:
```suggestion
            <FontAwesomeIcon icon={faCircleCheck} className="mr-2 h-4 w-4" />
```